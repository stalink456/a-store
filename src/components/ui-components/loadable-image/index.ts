export { LoadableImage } from './loadable-image';
