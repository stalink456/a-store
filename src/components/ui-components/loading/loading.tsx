import React from "react";
import { Spinner } from "@alfalab/core-components/spinner";

import styles from "./loading.module.css";

export const Loading: React.FC = () => {
  return (
    <div className={styles.loading} data-testid="loading">
      <Spinner visible={true} size="m" />
    </div>
  );
};
