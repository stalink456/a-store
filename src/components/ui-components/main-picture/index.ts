export { MainPicture } from "./main-picture";
