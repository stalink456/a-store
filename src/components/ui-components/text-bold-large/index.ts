export { TextBoldLarge } from "./text-bold-large";
