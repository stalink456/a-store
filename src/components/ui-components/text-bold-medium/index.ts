export { TextBoldMedium } from "./text-bold-medium";
