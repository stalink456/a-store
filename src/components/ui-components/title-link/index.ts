export { TitleLink } from './title-link';
