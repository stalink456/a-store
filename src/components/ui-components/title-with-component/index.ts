export { TitleWithComponent } from "./title-with-component";
