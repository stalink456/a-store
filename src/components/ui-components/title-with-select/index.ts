export { TitleWithSelect } from './title-with-select';
