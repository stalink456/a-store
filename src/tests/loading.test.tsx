import { render } from "@testing-library/react";
import { Loading } from "components/ui-components/loading";

describe("Loading component", () => {
  it("should render Loading component", () => {
    const { baseElement } = render(<Loading />);

    expect(baseElement).toMatchSnapshot();
  });
});
