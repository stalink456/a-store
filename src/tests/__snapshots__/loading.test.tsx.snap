// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should render Loading component 1`] = `
<body>
  <div>
    <div
      class="loading"
      data-testid="loading"
    >
      <svg
        class="spinner__spinner_1kljb spinner__component_1xxq0 spinner__m_1kljb spinner__visible_1kljb"
        fill="none"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <lineargradient
            id=":r0:_top"
            x1="0.05"
          >
            <stop
              offset="0.1"
              stop-color="currentColor"
              stop-opacity="0"
            />
            <stop
              offset="1"
              stop-color="currentColor"
              stop-opacity="0.3"
            />
          </lineargradient>
          <lineargradient
            id=":r0:_bottom"
            x1="0.05"
          >
            <stop
              offset="0"
              stop-color="currentColor"
              stop-opacity="1"
            />
            <stop
              offset="1"
              stop-color="currentColor"
              stop-opacity="0.3"
            />
          </lineargradient>
        </defs>
        <g
          stroke-width="4"
        >
          <path
            d="M6,24 A18,18 0 0 1 42,24"
            stroke="url(#:r0:_top)"
          />
          <path
            d="M42,24 A18,18 0 0 1 6,24"
            stroke="url(#:r0:_bottom)"
          />
          <path
            d="M6,24 A18,18 0 0 1 6 24"
            stroke="currentColor"
            stroke-linecap="round"
          />
        </g>
      </svg>
    </div>
  </div>
</body>
`;
